test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -976,6 +977,385 @@ public static CommandRunnerResult RunRestore(SimpleTestPathContext pathContext,
+                    net461);
+                var projectAPackages = Path.Combine(pathContext.SolutionRoot, "packages");
+
+                var externalRepositoryPath = pathContext.PackageSource;
+                var contosoRepositoryPath = pathContext.UserPackagesFolder;
+
+                var configPath = Path.Combine(pathContext.WorkingDirectory, "nuget.config");
+                SettingsTestUtils.CreateConfigurationFile(configPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""ExternalRepository"" value=""{externalRepositoryPath}"" />
+    <add key=""InternalRepository"" value=""{contosoRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""ExternalRepository"">
+            <namespace id=""Exter*"" />
+            <namespace id=""Other*"" />
+        </packageSource>
+        <packageSource key=""InternalRepository"">
+            <namespace id=""Conto*"" />   <!--Contoso.A package doesn't exist Contoso repository, so restore should fail-->
+            <namespace id=""Test*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+
+                var ExternalA = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",  // Initial version had package id conflict with Contoso repository
+                    Version = "1.0.0"
+                };
+                ExternalA.Files.Clear();
+                ExternalA.AddFile("lib/net461/externalA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalA);
+
+                var ExternalB = new SimpleTestPackageContext()
+                {
+                    Id = "External.B",  // name conflict resolved.
+                    Version = "2.0.0"
+                };
+                ExternalB.Files.Clear();
+                ExternalB.AddFile("lib/net461/externalB.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalB);
+
+                Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""Contoso.A"" version=""1.0.0"" targetFramework=""net461"" />
+  <package id=""External.B"" version=""2.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var result = RunRestore(pathContext, _failureExitCode);
+
+                // Assert
+                Assert.Contains("Unable to find version '1.0.0' of package 'Contoso.A'", result.Errors);
```

Comment:
How does the message look like?

What does it say? Does it indicate in which sources it was looked up?

Reply:
Here is sample:

```
NuGet Version: 6.0.0.20211
Restoring NuGet packages for solution C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\solution.sln.
MSBuild auto-detection: using msbuild version '17.0.0.30702' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
MSBuild P2P timeout [ms]: 120000
C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin\msbuild.exe "C:\Users\eryondon\AppData\Local\Temp\NuGetScratch\q5bvuc30.mo4.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:false /v:q /p:NuGetRestoreTargets="C:\Users\eryondon\AppData\Local\Temp\NuGetScratch\g1fcnqvd.3oz.nugetrestore.targets" /p:RestoreUseCustomAfterTargets="True" /p:RestoreTaskAssemblyFile="C:\NuGetProj\NuGet.Client\test\NuGet.Clients.FuncTests\NuGet.CommandLine.FuncTest\bin\Debug\net472\NuGet\NuGet.exe" /p:RestoreSolutionDirectory="C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\\" /p:SolutionDir="C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\\" /p:SolutionName="solution"
Restoring NuGet package Contoso.A.1.0.0.
Package namespace prefix matches found for package id 'Contoso.A' are: 'contosorepository'
Package namespace: Skipping source 'C:\Users\eryondon\.nuget\packages\' for package id 'Contoso.A'
Package namespace: Skipping source 'ExternalRepository' for package id 'Contoso.A'
Package namespace: Trying source 'ContosoRepository' for package id 'Contoso.A'
WARNING: Unable to find version '1.0.0' of package 'Contoso.A'.
  C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ContosoRepository: Package 'Contoso.A.1.0.0' is not found on source 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ContosoRepository'.
Restoring NuGet package External.B.2.0.0.
Package namespace prefix matches found for package id 'External.B' are: 'externalrepository'
Package namespace: Skipping source 'C:\Users\eryondon\.nuget\packages\' for package id 'External.B'
Package namespace: Trying source 'ExternalRepository' for package id 'External.B'
Package namespace: Skipping source 'ContosoRepository' for package id 'External.B'
Adding package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\packages'
Added package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\packages'
Added package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\packages' from source 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ExternalRepository'
WARNING: Unable to find version '1.0.0' of package 'Contoso.A'.
  C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ContosoRepository: Package 'Contoso.A.1.0.0' is not found on source 'C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ContosoRepository'.
NuGet Config files used:
    C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\NuGet.Config
    C:\NuGetProj\NuGet.Client\NuGet.Config
    C:\Users\eryondon\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
Feeds used:
    C:\Users\eryondon\.nuget\packages\
    C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ExternalRepository
    C:\NuGetProj\NuGet.Client\.test\work\4b725bb0\def48650\solution\ContosoRepository
Installed:
    1 package(s) to packages.config projects
```